### PR TITLE
10x faster Atom class

### PR DESCRIPTION
**Pull request by quantumsteve**
**Description of work.**

Using the proton number as an index leads to some substantial performance improvements.

I added two performance tests to show the difference

* test_z_performance: before: 0.246293s, after: 0.0214344s
* test_symbol_performance: before: 5.21779s, after: 0.310218s

Uses `tcb::span` suggested by @SimonHeybrock. Should that header be in a different place? 

**To test:**

* Code review
* I'd suggest running the performance tests with and without changes to `Atom.h` and Atom.cpp`

*There is no associated issue.*

I added a line to the release notes, though I suspect this is more about code smells than something observed by users.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by peterfpeterson** [comment]
Should `span.hpp` be grabbed through an external project?

**Comment by quantumsteve** [comment]
@peterfpeterson I moved span to an external project, which reduces the size of this PR.

**Comment by quantumsteve** [comment]
`DOWNLOAD_NO_EXTRACT` wasn't available until CMake v3.6
https://github.com/Kitware/CMake/commit/1ba9b535478a2a73100a6b0e9bf01983b4774cb8

File: docs/source/release/v4.1.0/framework.rst [hunk 1]
```diff
@@ -30,9 +30,9 @@ Improvements
 - Prevent an error due to the locale settings which may appear when reading, for instance, the incident energy Ei value from the logs in :ref:`ConvertUnits <algm-ConvertUnits>` and many other algorithms.
 - :code:`indices` and :code:`slicepoint` options have been added to :ref:`mantid.plots <mantid.plots>` to allow selection of which plane to plot from an MDHistoWorkspace. :code:`transpose` has also been added to transpose the axes of any 2D plot.
 - :ref:`Pseudo-Voigt <func-PseudoVoigt>` has been modified to be more in line with FULLPROF and GSAS.  One of its basic parameter, Height, is changed to Intensity.
+- 10x performance improvement in calls to Mantid::PhysicalConstants::getAtom.
```
Comment (peterfpeterson) [inline docs/source/release/v4.1.0/framework.rst]:
```suggestion
- 10x performance improvement in calls to ``Mantid::PhysicalConstants::getAtom``.
```

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
A very minor change to the release notes and this is good to go. Once this gets merged, you should announce the availability of `tcb/span.h` on slack.

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**